datasync/chaindatafetcher/kas/repository_contracts.go

```diff
@@ -0,0 +1,229 @@
+	"github.com/klaytn/klaytn/blockchain"
+	"github.com/klaytn/klaytn/common"
+	"strings"
+	"time"
+)
+
+// filterKIPContracts filters the deployed contracts to KIP7, KIP17 and others.
+func filterKIPContracts(api *api.PublicBlockChainAPI, event blockchain.ChainEvent) ([]*FT, []*NFT, []*Contract, error) {
+	var (
+		kip7s  []*FT
+		kip17s []*NFT
+		others []*Contract
+	)
+	caller := newContractCaller(api)
+	for _, receipt := range event.Receipts {
+		if receipt.ContractAddress == (common.Address{}) {
+			continue
+		}
+		blockNumber := event.Block.Number()
+		contract := receipt.ContractAddress
+		isKIP13, err := caller.isKIP13(contract, blockNumber)
+		if err != nil {
+			return nil, nil, nil, err
+		} else if !isKIP13 {
+			others = append(others, &Contract{Address: contract.Bytes()})
+			continue
+		}
+
+		if isKIP7, err := caller.isKIP7(contract, blockNumber); err != nil {
+			return nil, nil, nil, err
+		} else if isKIP7 {
+			kip7s = append(kip7s, &FT{Address: contract.Bytes()})
+			continue
+		}
+
+		if isKIP17, err := caller.isKIP17(contract, blockNumber); err != nil {
+			return nil, nil, nil, err
+		} else if isKIP17 {
+			kip17s = append(kip17s, &NFT{Address: contract.Bytes()})
+			continue
+		}
+		others = append(others, &Contract{Address: contract.Bytes()})
+	}
+	return kip7s, kip17s, others, nil
+}
+
+// InsertContracts inserts deployed contracts in the given chain event into KAS database.
+func (r *repository) InsertContracts(event blockchain.ChainEvent) error {
+	kip7s, kip17s, others, err := filterKIPContracts(r.blockchainApi, event)
+	if err != nil {
+		logger.Error("Failed to filter KIP contracts", "err", err, "blockNumber", event.Block.NumberU64())
+		return err
+	}
+
+	if err := r.insertFTs(kip7s); err != nil {
+		logger.Error("Failed to insert KIP7 contracts", "err", err, "blockNumber", event.Block.NumberU64(), "numKIP7s", len(kip7s))
+		return err
+	}
+
+	if err := r.insertNFTs(kip17s); err != nil {
+		logger.Error("Failed to insert KIP17 contracts", "err", err, "blockNumber", event.Block.NumberU64(), "numKIP17s", len(kip17s))
+		return err
+	}
+
+	if err := r.insertContracts(others); err != nil {
+		logger.Error("Failed to insert other contracts", "err", err, "blockNumber", event.Block.NumberU64(), "numContracts", len(others))
+		return err
+	}
+
+	return nil
+}
+
+// insertContracts inserts the contracts which are divided into chunkUnit because of max number of placeholders.
+func (r *repository) insertContracts(contracts []*Contract) error {
+	chunkUnit := maxPlaceholders / placeholdersPerContractItem
+	var chunks []*Contract
+
+	for contracts != nil {
+		if placeholdersPerContractItem*len(contracts) > maxPlaceholders {
+			chunks = contracts[:chunkUnit]
+			contracts = contracts[chunkUnit:]
+		} else {
+			chunks = contracts
+			contracts = nil
+		}
+
+		if err := r.bulkInsertContracts(chunks); err != nil {
+			//log.Logger.Sugar().Errorw("failed to bulkInsertContracts", "err", err, "itemsCount", len(chunks))
```

Comment:
Did you intend this?

Reply:
Removed it. 32fa023af377a36886c823547167aefa9cacb049